src/user_database.ts

```diff
@@ -29,7 +31,8 @@ export interface UserDatabase {
   dropSchema(): Promise<void>;
 }
 
-type UserDatabaseTransaction<T extends any[], R> = (ctxt: UserDatabaseClient, ...args: T) => Promise<R>;
+type UserDatabaseQuery<C extends UserDatabaseClient, R, T extends unknown[]> = (ctxt: C, ...args: T) => Promise<R>;
```

Comment:
What is the downside of forcing the use of `UserDatabaseTransaction` for all DB queries? Overheads?

Reply:
Ideally we want to force UserDatabaseQuery to be read-only, so it cannot modify the database.